src/manifest.rs

```diff
@@ -183,6 +164,75 @@ impl Manifest {
         data.parse()
     }
 
+    /// Get the specified table from the manifest.
+    pub fn get_table<'a>(
+        &'a mut self,
+        table_path: &[String],
+    ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+
+        /// Descend into a manifest until the required table is found.
+        fn descend<'a>(
+            input: &'a mut BTreeMap<String, toml::Value>,
+            mut path: VecDeque<&String>,
```

Comment:
Interesting, using sub slices is pretty clever :)

(But don't go out your way to optimize stuff like this; this whole project is in no way performance critical code!)